src/v/metrics/CMakeLists.txt

```diff
@@ -0,0 +1,10 @@
+v_cc_library(
+  NAME
+    metrics
+  HDRS
+    "metrics_registry.h"
+  SRCS
+    metrics_registry.cc
```

Comment:
Now that we have a metrics library, seems like the ssx metrics stuff should live here?

Reply:
Happy to move, I am not entirely sure what qualifies as "ssx" and what doesn't. The `ssx::metric_groups` is strictly speaking still an extension to seastar so it belongs into ssx? I have no strong opinion though.

Reply:
Yeah ssx is an extension to seastar, also I think there was a desire to keep it header only. I think it only makes sense to put stuff there that is plausibly something that would be a useful generic addition to seastar, i.e., something that would reasonably be accepted upstream (regardless if we try or not) or form the basis for a "standard library for seastar" for general use.

This metrics stuff including the registry seems RP specific (e.g., since it embeds the split behind internal/public metrics, etc). Furthermore it doesn't seem like it is naturally header only.

So I vote to move it.

Reply:
Ok I will move the file and remove `ssx` from the namespace.

Reply:
Have done the change. Let me know whether it matches what you had in mind. Then I will do a rebase and split it up properly into the respective commits.